jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/UpdateChecker.java

```diff
@@ -139,11 +150,17 @@ void setDisableUpdateCheck(boolean disableUpdateCheck) {
       }
 
       // Check for update
-      HttpURLConnection connection = (HttpURLConnection) new URL(versionUrl).openConnection();
+      FailoverHttpClient httpClient = new FailoverHttpClient(true, false, log);
+      Response response =
+          httpClient.get(
```

Comment:
You can put this line into `try`, can't you? Otherwise, if `httpClient.get()` throws an exception, `httpClient.shutDown()` is not called.